lib/hsa/kernels/sum_hfb.cl

```diff
@@ -20,23 +20,26 @@ __kernel void sum_hfb(__global float *data, __constant uint *compressed_lost_sam
 
   for(int sample=0; sample<num_samples; sample+=4) {
 
-      if(!compressed_lost_samples_buf[sample]) {
-
-          // Load data into vectors
-          data_1.s0 = data[1024*NUM_SUB_FREQS*sample + beam*NUM_SUB_FREQS + freq];
-          data_1.s1 = data[1024*NUM_SUB_FREQS*(sample + 1) + beam*NUM_SUB_FREQS + freq];
-          data_1.s2 = data[1024*NUM_SUB_FREQS*(sample + 2) + beam*NUM_SUB_FREQS + freq];
-          data_1.s3 = data[1024*NUM_SUB_FREQS*(sample + 3) + beam*NUM_SUB_FREQS + freq];
-          
-          data_2.s0 = data[1024*NUM_SUB_FREQS*sample + beam*NUM_SUB_FREQS + freq + 1];
-          data_2.s1 = data[1024*NUM_SUB_FREQS*(sample + 1) + beam*NUM_SUB_FREQS + freq + 1];
-          data_2.s2 = data[1024*NUM_SUB_FREQS*(sample + 2) + beam*NUM_SUB_FREQS + freq + 1];
-          data_2.s3 = data[1024*NUM_SUB_FREQS*(sample + 3) + beam*NUM_SUB_FREQS + freq + 1];
+     // Mask data from lost samples.
+     const float mask_0 = 1.f - compressed_lost_samples_buf[sample];
+     const float mask_1 = 1.f - compressed_lost_samples_buf[sample + 1];
+     const float mask_2 = 1.f - compressed_lost_samples_buf[sample + 2];
+     const float mask_3 = 1.f - compressed_lost_samples_buf[sample + 3];
```

Comment:
I would leave this as an integer subtraction `float mask_0 = 1 - compressed_lost_samples_buf[sample]` just too avoid any float subtraction oddities - plus I think it's slightly faster to do an int op here. 

Also I don't think you can write this as a `const`, since you do update it in the loop.  (although maybe the redeclaration might allow it...)